lib/sufia/import/generic_file_translator.rb

```diff
@@ -0,0 +1,45 @@
+module Sufia::Import
+  # Imports a Sufia 6.0-exported GenericFile into a Sufia PCDM GernericWork and FileSet
+  class GenericFileTranslator
+    attr_accessor :verbose
+
+    # @param Hash settings needed to pass through to builder
+    #        @attr sufia6_user      - User name for accessing the sufia 6 fedora
+    #        @attr sufia6_password  - Password for accessing the sufia 6 fedora
+    #        @attr import_binary    - (default true) Import the binary content from fedora
+    def initialize(settings)
+      # TODO: work builder doesn't need settings; just file builder
+      @work_builder = WorkBuilder.new(settings)
+      @file_set_builder = FileSetBuilder.new(settings)
+      @verbose = false
+    end
+
+    # loop through the files found in import_dir and map each GenericFile to a Sufia.primary_work and FileSet
+    # @param import_dir a directory containing export files, e.g.: generic_file_[id].json
+    # @return list of imported files
+    def import(import_dir, filename_prefix)
+      filename_prefix ||= "generic_file_"
+      raise "No such directory: '#{import_dir}'" unless Dir.exist?(import_dir)
+      # get filenames
+      files = Dir.glob(File.join(import_dir, '*')).select { |f| exported_gf?(f, filename_prefix) }
+      files.each { |f| import_file(f) }
+    end
+
+    private
+
+      def import_file(filename)
+        puts "Importing #{File.basename(filename)}" if @verbose
```

Comment:
Thanks for looking at it. You are probably right. I was hesitant, though, because the class is primarily intended to be used by the exe. maybe there is a way for the exe to print from the rails log if it's in verbose mode? then i can remove the 'verbose' stuff from here and just write to the log straight-up.